src/snowflake/connector/cursor.py

```diff
@@ -120,10 +121,10 @@
 class ResultMetadata(NamedTuple):
     name: str
     type_code: int
-    display_size: int
-    internal_size: int
-    precision: int
-    scale: int
+    display_size: Optional[int]
+    internal_size: Optional[int]
+    precision: Optional[int]
+    scale: Optional[int]
```

Comment:
We should be using future annotations
```suggestion
    display_size: int | None
    internal_size: int | None
    precision: int | None
    scale: int | None
```